### ISSUE TITLE
[translatable] :   Specified key was too long; max key length is 767 bytes

### ISSUE THREAD

**Issue by skytorner** [KnpLabs/DoctrineBehaviors#391]
Hi,

 

I'm doing a SF4 tutorial.

I use this bundle in SF2 project and it works fine but in SF4, I cant generate the translation table because of the following error : 

`An exception occurred while executing 'CREATE TABLE category_translation (id INT AUTO_INCREMENT NOT NULL, translatable_id INT DEFAULT NULL, name VARCHAR(255) NOT NULL, description VARCHAR(255) NOT NULL, locale VARCHAR(255) NOT NULL, INDEX IDX_3F207042C2AC5D3 (translatable_id), UNIQUE INDEX category_translation_unique_translation (translatable_id, locale), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = InnoDB': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes`

If I understand the error, I have to increase the length of the KEY to be able to update the DB.

My question is : which key am I supposed to increase the length, and how can I do it ? (via code ? PHPMyAdmin ?)

**Comment by roenschg** [comment]
@skytorner you can find an explanation to fix it here: https://blog.joeymasip.com/gedmo-translations-in-symfony-4/

TLDR; Use MySQL Version 5.7 or change the charset because a key of 255 will us 1020 Bytes.
